addons/website_livechat/models/mail_channel.py

```diff
@@ -53,23 +52,14 @@ def _get_visitor_history(self, visitor):
         recent_history = self.env['website.track'].search([('page_id', '!=', False), ('visitor_id', '=', visitor.id)], limit=3)
         return ' → '.join(visit.page_id.name + ' (' + visit.visit_datetime.strftime('%H:%M') + ')' for visit in reversed(recent_history))
 
-    def close_livechat_request_session(self, type='leave', **kwargs):
-        """ Set deactivate the livechat channel and notify (the operator) the reason of closing the session."""
-        self.ensure_one()
-        if self.livechat_active:
-            self.livechat_active = False
-            # avoid useless notification if the channel is empty
-            if not self.channel_message_ids:
-                return
-            # Notify that the visitor has left the conversation
-            name = _('The visitor') if not self.livechat_visitor_id else self.livechat_visitor_id.display_name
-            if type == 'cancel':
-                message = _('has started a conversation with %s. The chat request has been canceled.') % kwargs.get('speaking_with', 'an operator')
-            else:
-                message = _('has left the conversation.')
-            leave_message = '%s %s' % (name, message)
-            self.message_post(author_id=self.env.ref('base.user_root').sudo().partner_id.id,
-                              body=leave_message, message_type='comment', subtype_xmlid='mail.mt_comment')
+    def _get_visitor_leave_message(self, **kwargs):
+        name = _('The visitor') if not self.livechat_visitor_id else self.livechat_visitor_id.display_name
```

Comment:
Maybe we could somehow improve string formating in order to ease translations, like
```
_('%s has started a conversation') % blah
_('%s has left the conversation') % blah
```

By the way using kwargs to give stuff to the method is a bit strange. It would be better to have explicit parameters while we are modifying its code ...